src/libponyc/type/cap.c

```diff
@@ -251,28 +238,25 @@ bool is_cap_sub_cap_bound(token_id sub, token_id subalias, token_id super,
       return false;
   }
 
-  if((sub == super) || (super == TK_TAG) || (super == TK_CAP_ANY_BIND))
+  if((sub == super) || (super == TK_TAG) || (super == TK_CAP_ANY))
     return true;
 
-  // Sub and super share the same initial bounds. Some instantiation of the
-  // sub rcap must be a subtype of every instantiation of the super rcap.
+  // Sub and super share the same initial bounds. Every instantiation of the
+  // super rcap must be a supertype of some instantiation of the sub rcap.
```

Comment:
Note the subtle difference in the description here, which actually does change the semantics.

This distinction was actually the bulk of the bug, in my opinion, so please check my logic here carefully, as this is a part of the fix that I discovered on my own and didn't discuss with you in our last conversation about this issue.

I basically rewrote the entire switch-case from scratch here.  It is very similar to the semantics of `is_cap_sub_cap`, except some extra cases are allowed when comparing gen-caps to gen-caps.

This is to be expected given the semantics I chose in this description, because the real only distinction from `is_cap_sub_cap` is the use of "some" instead of "every" in reference to the instantiations of the sub rcap, and this is a distinction without a difference when there is only one possible instantiation of the sub rcap.